core/include/moveit/task_constructor/stages/generate_place_pose.h

```diff
@@ -54,6 +54,9 @@ class GeneratePlacePose : public GeneratePose {
 
 	void setObject(const std::string &object) { setProperty("object", object); }
 
+	void setAugmentRotations(bool augment_rotations) { setProperty("augment_rotations",
```

Comment:
I don't like the property name yet. What about `consider_symmetry`?
Additionally to this boolean flag, it would be nice to also set the number of z rotations to use.

Reply:
How about a single property then, that defaults to 1 or 0?

Reply:
Exactly. We need both parameters independently.

Reply:
I'm inclined to make the same argument as in #98 here then.
Should `GeneratePlacePose` be a general-purpose stage that handles that many alternatives?
You can always plug together multiple of them with a parallel container :sunglasses:

Reply:
I think, it would be great to have:
- a generic `CartesianPose` generator, which just creates a fixed pose relative to another frame (much like `FixedState`)
- a generic `VaryPose` generator that wraps CartesianPose and allows to augment different variations of the original pose(s) received from the wrapped stage.

Of course, one could also combine both stages. In any case, both GenerateGraspPose and GeneratePlacePose could be replaced by (or based on) this more generic stage.

Reply:
> I'm inclined to make the same argument as in #98 here then.
> Should `GeneratePlacePose` be a general-purpose stage that handles that many alternatives?
> You can always plug together multiple of them with a parallel container

`GeneratePlacePose` could be a simple Cartesian pose generator and we could add a wrapper that samples for alternative orientations of the object. I think it make sense to provide a lot of primitive stages (i.e. CartesianPose) but also keep high-level versions for convenience and semantics.